src/Stratis.Bitcoin.Features.PoA/PoABlockHeaderValidator.cs

```diff
@@ -0,0 +1,50 @@
+﻿using Microsoft.Extensions.Logging;
+using NBitcoin;
+using NBitcoin.Crypto;
+
+namespace Stratis.Bitcoin.Features.PoA
+{
+    public class PoABlockHeaderValidator
+    {
+        private readonly ILogger logger;
+
+        public PoABlockHeaderValidator(ILoggerFactory factory)
+        {
+            this.logger = factory.CreateLogger(this.GetType().FullName);
+        }
+
+        public void Sign(Key key, PoABlockHeader header)
+        {
+            ECDSASignature signature = key.Sign(header.GetHash());
+
+            header.BlockSignature = new BlockSignature { Signature = signature.ToDER() };
+        }
+
+        public bool VerifySignature(PubKey pubKey, PoABlockHeader header)
+        {
+            if (header.BlockSignature.IsEmpty())
```

Comment:
maybe check for null as well ` if (header.BlockSignature == null || header.BlockSignature.IsEmpty())`

Reply:
fixed